src/app/modules/user/user-list.page.tsx

```diff
@@ -13,25 +15,37 @@ interface UserData {
 }
 
 export function UserListPage() {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [errorMsg, setErrorMsg] = useState('');
+  const currentOffset = getCurrentOffset(searchParams);
   const { loading, error, data } = useQuery(UserListQuery, {
-    variables: { offset: 0, limit: 20 },
+    variables: { offset: currentOffset, limit: PAGINATION_LIMIT },
   });
   const navigate = useNavigate();
 
   useEffect(() => {
     if (error?.graphQLErrors[0].name == 'UnauthorizedError') {
       navigate('/login');
+    } else if (error) {
+      setErrorMsg(error.message);
+    } else {
+      setErrorMsg('');
```

Comment:
```suggestion
    } else if (error) {
      setErrorMsg(error.message);
    }
```
(precisa do último else? 🤔 )